src/app/pages/EarnTez/components/baker-card.tsx

```diff
@@ -0,0 +1,165 @@
+import { useStakedAmount } from 'app/hooks/use-baking-hooks';
+import { ReactComponent as OutLinkIcon } from 'app/icons/base/outLink.svg';
+import { T, t, toShortened } from 'lib/i18n';
+import { getTezosGasMetadata } from 'lib/metadata';
+import { Baker, useKnownBaker } from 'lib/temple/front';
+import { useBooleanState } from 'lib/ui/hooks';
+import { toPercentage } from 'lib/ui/utils';
+import { useOnTezosBlock } from 'temple/front';
+import { useBlockExplorerHref } from 'temple/front/use-block-explorers';
+import { TezosNetworkEssentials } from 'temple/networks';
+import { TempleChainKind } from 'temple/types';
+
+import { getBakerAddress } from '../utils';
+
+import { BakerAvatar } from './baker-avatar';
+import { BakerName } from './baker-name';
+import { StakingCard } from './staking-card';
+
+interface PropComponentProps {
+  /** Atomic value */
+  staked: number;
+}
+
+interface BakerCardProps {
+  className?: string;
+  network: TezosNetworkEssentials;
+  accountPkh: string;
+  baker: string | Baker;
+  metricsType?: 'delegation' | 'staking';
+  ActionButton?: React.ComponentType<PropComponentProps>;
+  HeaderRight?: React.ComponentType<PropComponentProps>;
+  onClick?: SyncFn<string | Baker>;
+}
+
+export const BakerCard = memo(
+  ({
+    className,
+    network,
+    accountPkh,
+    ActionButton,
+    HeaderRight,
+    metricsType = 'delegation',
+    baker: bakerOrAddress,
+    onClick
+  }: BakerCardProps) => {
+    const { rpcBaseURL, chainId } = network;
+
+    const [hovered, setHovered, setUnhovered] = useBooleanState(false);
+    const { data: stakedData, mutate } = useStakedAmount(rpcBaseURL, accountPkh);
+    const bakerPkh = getBakerAddress(bakerOrAddress);
+    const bakerFromProps = typeof bakerOrAddress === 'object' ? bakerOrAddress : undefined;
+    const explorerHref = useBlockExplorerHref(TempleChainKind.Tezos, chainId, 'address', bakerPkh);
+    const { data: baker = bakerFromProps } = useKnownBaker(
+      bakerFromProps ? null : bakerPkh,
+      network.chainId,
+      !bakerFromProps
+    );
+
+    useOnTezosBlock(rpcBaseURL, () => void mutate());
+
+    const { symbol } = getTezosGasMetadata(chainId);
+
+    const stakedAtomic = useMemo(() => stakedData?.toNumber() || 0, [stakedData]);
+
+    const headerRight = useMemo(
+      () => HeaderRight && <HeaderRight staked={stakedAtomic} />,
+      [HeaderRight, stakedAtomic]
+    );
+
+    const bakerAvatar = <BakerAvatar className="mr-2" address={baker?.address} bakerName={baker?.name} />;
+    const bakerName = <BakerName>{baker ? baker.name : <T id="unknownBakerTitle" />}</BakerName>;
+
+    const delegationFeePercentage = useMemo(() => (baker ? toPercentage(baker.delegation.fee) : '-'), [baker]);
+    const stakingFeePercentage = useMemo(() => (baker ? toPercentage(baker.staking.fee) : '-'), [baker]);
+    const stakingApyPercentage = useMemo(() => (baker ? toPercentage(baker.staking.estimatedApy) : '-'), [baker]);
```

Comment:
you could get rid of this baker defined check like this:
`export const toPercentage = (value?: BigNumber.Value, fallback = '-', decimalPlaces = 2) =>
  value ? `${toLocalFormat(new BigNumber(value).times(100), { decimalPlaces })}%` : fallback;`